flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlAlterTableAdd.java

```diff
@@ -0,0 +1,150 @@
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.sql.parser.ddl;
+
+import org.apache.flink.sql.parser.ddl.constraint.SqlTableConstraint;
+import org.apache.flink.sql.parser.ddl.position.SqlTableColumnPosition;
+
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * SqlNode to describe ALTER TABLE table_name ADD column/constraint/watermark clause.
+ *
+ * <p>Example: DDL like the below for add column/constraint/watermark.
+ *
+ * <pre>{@code
+ * -- add single column
+ * ALTER TABLE mytable ADD new_column STRING COMMENT 'new_column docs';
+ *
+ * -- add multiple columns, constraint, and watermark
+ * ALTER TABLE mytable ADD (
+ *     log_ts STRING COMMENT 'log timestamp string' FIRST,
+ *     ts AS TO_TIMESTAMP(log_ts) AFTER log_ts,
+ *     PRIMARY KEY (id) NOT ENFORCED,
+ *     WATERMARK FOR ts AS ts - INTERVAL '3' SECOND
+ * );
+ * }</pre>
+ */
+public class SqlAlterTableAdd extends SqlAlterTable {
+
+    // Whether the column is added by a paren, currently it is only used for SQL unparse.
+    private final boolean withParen;
+    private final SqlNodeList addedColumns;
+    @Nullable private final SqlWatermark watermark;
+    private final List<SqlTableConstraint> constraint;
+
+    public SqlAlterTableAdd(
+            SqlParserPos pos,
+            SqlIdentifier tableName,
+            boolean withParen,
+            SqlNodeList addedColumns,
+            @Nullable SqlWatermark sqlWatermark,
+            List<SqlTableConstraint> constraint) {
+        super(pos, tableName, null);
+        this.withParen = withParen;
+        this.addedColumns = addedColumns;
+        this.watermark = sqlWatermark;
+        this.constraint = constraint;
+    }
+
+    /** Returns the column constraints plus the table constraints. */
+    public List<SqlTableConstraint> getFullConstraints() {
```

Comment:
is the method needed? Seems it's nerver used.

Reply:
remove it first